p4-16/spec/P4-16-spec.mdk

```diff
@@ -4209,9 +4217,9 @@ expressions are legal:
   The `index` is an expression that must be one of the following types:
 
   - `int` - an arbitrary-precision integer (Section [#sec-arbitrary-precision-integers])
-  - `bit<W>` - a `W`-bit unsigned integer where W≥0 (Section [#sec-unsigned-integers])
-  - `int<W>` - a `W`-bit signed integer where W≥1 (Section [#sec-signed-integers])
-  - a serializable `enum` with an underlying type that is `bit<W>` or `int<W>` where W≥1 (Section [#sec-enum-types]).
```

Comment:
This condition is only correct for signed integers as the underlying type of an enum and not for unsigned integers. It might be better to discuss when these types are well-formed (the conditions of their width holds) and state that when a type is used anywhere it's assumed that it is well-formed so we don't have to keep repeating these conditions.